packages/interaction-viewer/src/interaction-viewer.ts

```diff
@@ -0,0 +1,177 @@
+
+import styles from "./styles";
+import { APIInteractionData } from "./data";
+// eslint-disable-next-line import/no-duplicates
+import InteractionTooltip from "./interaction-tooltip";
+// eslint-disable-next-line import/no-duplicates
+import { FILTER_SELECT } from "./interaction-filters";
+
+// Import additional components
+// eslint-disable-next-line import/no-duplicates
+import "./interaction-filters";
+// eslint-disable-next-line import/no-duplicates
+import "./interaction-tooltip";
+
+const dispatchLoadedEvent = (el: HTMLElement, error?: string) => {
+  el.dispatchEvent(
+    new CustomEvent("protvista-event", {
+      detail: {
+        loaded: true,
+        error,
+      },
+      bubbles: true,
+    })
+  );
+};
+
+const filterAdjacencyMap = (
+  adjacencyMap: AdjacencyMap,
+  filteredAccessions: string[]
+): AdjacencyMap => {
+  if (!filteredAccessions) {
+    return adjacencyMap;
+  }
+  return adjacencyMap.map(({ accession, interactors }) => ({
+    accession,
+    interactors: interactors.filter(
+      (interactor) =>
+        filteredAccessions.includes(interactor) ||
+        filteredAccessions.includes(accession)
+    ),
+  }));
+};
+
+@customElement("interaction-viewer")
+export default class InteractionViewer extends LitElement {
+  private nodes: APIInteractionData[] = null;
+
+  @property({ reflect: true })
+  accession: string;
+
+  @state()
+  private processedData?: ProcessedData;
+
+  @state()
+  private filteredAccessions?: string[];
+
+  private handleFilterSelection(event: CustomEvent): void {
+    if (event.detail.length === 0) {
+      // Reset filters
+      this.filteredAccessions = undefined;
+    } else if (
+      !this.filteredAccessions ||
+      this.filteredAccessions.length === 0
+    ) {
+      // First filter applied
+      this.filteredAccessions = event.detail;
+    } else {
+      // AND additional filters
+      this.filteredAccessions = _intersection(
+        event.detail,
+        this.filteredAccessions
+      );
+    }
+  }
+
+  async connectedCallback(): Promise<void> {
+    super.connectedCallback();
+    this.addEventListener(FILTER_SELECT, this.handleFilterSelection);
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.removeEventListener(FILTER_SELECT, this.handleFilterSelection);
+  }
+
+  static styles = styles;
+
+  getNodeByAccession(accession: string): APIInteractionData {
+    return this.nodes.find((node) => node.accession === accession);
+  }
+
+  async updated(changedProperties: Map<string, any>): Promise<void> {
+    // Only run this if it's the accession that has changed
+    if (Array.from(changedProperties.keys()).includes("accession")) {
+      if (!this.accession) {
+        return;
+      }
+      const response = await load(
+        `https://www.ebi.ac.uk/proteins/api/proteins/interaction/${this.accession}.json`
+      );
+      const data = response.payload as APIInteractionData[];
+      this.processedData = process(data);
+    }
```

Comment:
Issue with async is that we might start to get race conditions

```suggestion
  updated(changedProperties: Map<string, any>): Promise<void> {
    // Only run this if it's the accession that has changed
    if ('accession' in changedProperties) {
      if (!this.accession) {
        return;
      }
      // fetch then later on update the state
      load(
        `https://www.ebi.ac.uk/proteins/api/proteins/interaction/${this.accession}.json`
      ).then((response) => {
        // eventually update the state
        this.processedData = process(response.payload as APIInteractionData[]);
      });
      // but first remove any previous data from the state (effectively, we are in a loading state)
      this.processedData = null;
    }
```